src/test/java/org/kpmp/geneExpression/RTGeneExpressionDataServiceTest.java

```diff
@@ -32,16 +32,51 @@ public void tearDown() throws Exception {
     }
 
     @Test
-    public void testGetByComparisonTypeGeneSymbolAndTissueType() throws Exception {
-        RTExpressionDataAllSegments aData = new RTExpressionDataAllSegments();
-        List<RTExpressionDataAllSegments> aDataList = Arrays.asList(aData);
-        RTExpressionDataGTI gData = new RTExpressionDataGTI();
-        List<RTExpressionDataGTI> gDataList = Arrays.asList(gData);
-        when(rtExpressionDataAllSegmentsRepository.findByGeneSymbol("gene")).thenReturn(aDataList);
-        when(rtExpressionDataGTIRepository.findByGeneSymbol("gene")).thenReturn(gDataList);
-        assertEquals(aDataList, rtExpressionDataService.getByComparisonTypeAndGeneSymbol("all_segments", "gene"));
-        assertEquals(gDataList, rtExpressionDataService.getByComparisonTypeAndGeneSymbol("glom_tub", "gene"));
-        assertEquals(null, rtExpressionDataService.getByComparisonTypeAndGeneSymbol("blah", "gene"));
+    public void testGetByComparisonTypeAndGeneSymbolPerTissue() throws Exception {

```

Comment:
This should really be broken out into multiple tests. There is a LOT of setup, and I think it could be much simpler to follow if it were individual tests.

Reply:
The setup for this test is so complicated because the service returns an object containing the results of a bunch of queries and needs to test that each fields contains the correct data, but in reality it's only testing one method. I don't think breaking this up makes sense.